File: backend/rest/src/main/java/com/github/thiagogarbazza/training/springangular/rest/comum/CustomPageResource.java
package com.github.thiagogarbazza.training.springangular.rest.comum;

import com.github.thiagogarbazza.simplemessage.SimpleMessage;
import lombok.Getter;
import lombok.Setter;

import java.util.Collection;
import java.util.HashMap;
import java.util.Map;

@Getter
@Setter
public class CustomPageResource<T> {

  private Collection<T> content;
  private Integer contentSize;
  private boolean hasContent;
  private boolean hasNext;
  private boolean hasPrevious;
  private boolean isFirst;
  private boolean isLast;
  private Collection<SimpleMessage> messages;
  private Integer pageNext;
  private Integer pageNumber;
  private Integer pagePrevious;
  private Integer pageSize;
  private Map<String, Object> parameters = new HashMap<>();
  private Long totalElements;
  private Integer totalPages;
}

File: front-end/src/shared/models/custom-page-resource.ts
import { SimpleMessage } from '@shared/components/simple-messages/simple-message';
import { Ordenacao } from '@shared/components/paginacao/ordenacao';

export class CustomPageResource<T> {

  content: Array<T>;
  contentSize: number;
  hasContent: boolean;
  hasNext: boolean;
  hasPrevious: boolean;
  isFirst: boolean;
  isLast: boolean;
  messages: Array<SimpleMessage>;
  pageNext: number;
  pageNumber: number;
  pagePrevious: number;
  pageSize: number;
  parameters: any;
  totalElements: number;
  totalPages: number;
  sort: Ordenacao;
}




